integration-tests/test_hub.py

```diff
@@ -137,4 +138,41 @@ def test_hub_up():
             '-u', 'jupyterhub',
             '--no-pager'
         ])
-        raise
\ No newline at end of file
+        raise
+
+
+@pytest.mark.asyncio
+async def test_idle_culler  ():
+    """
+    User logs in, starts a server & stays idle for 1 min
+    """
+    # This *must* be localhost, not an IP
+    # aiohttp throws away cookies if we are connecting to an IP!
+    hub_url = 'http://localhost'
+    username = secrets.token_hex(8)
+
+    assert 0 == await (await asyncio.create_subprocess_exec(*TLJH_CONFIG_PATH, 'set', 'auth.type', 'dummyauthenticator.DummyAuthenticator')).wait()
+    # Check every 10s for idle servers to cull
+    assert 0 == await (await asyncio.create_subprocess_exec(*TLJH_CONFIG_PATH, 'set', 'services.cull.every', "10")).wait()
+    # Apart from servers, also cull users
+    assert 0 == await (await asyncio.create_subprocess_exec(*TLJH_CONFIG_PATH, 'set', 'services.cull.users', "True")).wait()
+    # Cull servers and users after 60s of activity
+    assert 0 == await (await asyncio.create_subprocess_exec(*TLJH_CONFIG_PATH, 'set', 'services.cull.max_age', "60")).wait()
+    assert 0 == await (await asyncio.create_subprocess_exec(*TLJH_CONFIG_PATH, 'reload')).wait()
+
+    async with User(username, hub_url, partial(login_dummy, password='')) as u:
+            await u.login()
+            # Start user's server
+            await u.ensure_server()
+            # Assert that the user exists
+            assert pwd.getpwnam(f'jupyter-{username}') is not None
+
+            # Check that we can get to the user's server
+            r = await u.session.get(u.hub_url / 'hub/api/users' / username,
+                headers={'Referer': str(u.hub_url / 'hub/')})
+            assert r.status == 200
+            time.sleep(60)
```

Comment:
Since we're in an async function, `await asyncio.sleep(60)` is probably more appropriate.

Reply:
@yuvipanda, I just noticed this test is failing intermittently :disappointed:
Do you think it would be wiser to use [exponential_backoff](https://github.com/jupyterhub/jupyterhub/blob/af0082a16ba365a4b1bfb79f9d01a7c1a6cee5be/jupyterhub/utils.py#L93) here with maybe an 100s timeout, instead of relying on the 60s sleep to be enough for the server to be culled?

Reply:
Ah yeah, that makes sense! I think we should have two tests:

1. One with a 100s timeout to make sure the server is culled
2. Another, with a 30s timeout, to make sure the server is *not* culled

I think this will give us a lot more confidence in the code. what do you think?

Reply:
This sounds perfect :D